MPI/isx.c

```diff
@@ -97,8 +99,17 @@ static char * parse_params(const int argc, char ** argv)
   MAX_KEY_VAL = DEFAULT_MAX_KEY;
   NUM_BUCKETS = NUM_PES;
   BUCKET_WIDTH = (uint64_t) ceil((double)MAX_KEY_VAL/NUM_BUCKETS);
-  char * log_file = argv[2];
   char scaling_msg[64];
+  char * log_file;
+
+  if(argc == 3) {
+    NUM_ITERATIONS = 1u;
+    log_file = argv[2];
+  }
+  if(argc == 4) {
+    NUM_ITERATIONS = (uint64_t) strtoull(argv[2], NULL, 10);
+    log_file = argv[3];
+  }
```

Comment:
Not proposing a change, but it would have also been fine to keep the original code structure.
```
  NUM_ITERATIONS = 1u;
  log_file = argv[2];
  
  if(argc == 4) {
    NUM_ITERATIONS = (uint64_t) strtoull(argv[2], NULL, 10);
    log_file = argv[3];
  }
```
This has the advantage that there is a default behavior when argc is not 3 or 4. Of course it has to be one of those values today, but we may change the code again in the future and the argc checking would break. You could also use `if ... else if ... else` to accomplish something similar.

Reply:
@kholland-intel I think I would also choose the `if - else if - else` block here. Other than this, everything else looks good to me.

Reply:
I hope I understood you guys' comment correctly, does this latest commit reintroduce that default behavior as expected?